integration-tests/src/server.js

```diff
@@ -20,26 +21,44 @@ const app = express()
 const server = http.createServer(app)
 server.setTimeout(serverTimeout)
 
+const firestore = new Firestore()
+
 const testTimeout = async timeout => {
   await delay(timeout)
   throw new Error(`Test timeout after ${timeout}ms`)
 }
 
+
+
 const runTest = fn => withPuppeteer(async ({ browser, context, name, page, req, timeout }) => {
   const targetEnvParams = { ...envs[req.query.targetEnv] }
+  const testGroup = req.query.testGroup
+  const startTime = new Date()
+  const host = req.headers.host
+  const reportTest = async error => {
+    const endTime = new Date()
+    const testReport = { testGroup, runtimeInMilliseconds: endTime - startTime, status: error ? 'error' : 'success', testName: name, error, host, startTime }
+    await firestore.collection('tests').doc().create(testReport)
+  }
+  if (host === 'terra-bueller.appspot.com') {
+    throw new Error('Do not call without a version specified')
+  }
   try {
     const test = withScreenshot(name, fn)
     const result = await Promise.race([
       test({ browser, context, page, ...targetEnvParams }),
       testTimeout(timeout)
```

Comment:
TOL: We may want to consider recording test timeout events in a distinct way to make it easier to report on how often this happens. Could be particularly interesting for load tests.

Reply:
I also thought about this, but we won't catch timeouts within the test, only global timeouts on the entire run (on talking to Brett), so the inconsistency may hurt more than help.